include/contractor/graph_contractor.hpp

```diff
@@ -47,7 +47,7 @@ class GraphContractor
                            bool shortcut,
                            bool forward,
                            bool backward)
-            : weight(weight), id(id), originalEdges(std::min((unsigned)1 << 28, original_edges)),
+            : weight(weight), id(id), originalEdges(std::min((1u << 28) - 1u, original_edges)),
```

Comment:
cc @MoKob who is refactoring this code right now anyway - from a first sight it looks like you're changing semantic here - why?